### ISSUE TITLE
[engineering/training-receipt] Define training-run manifest schema (separate from inference RunManifest)

### ISSUE THREAD

**Issue by moapacha** [p-to-q/wittgenstein#482]
## Context

The inference-time \`RunManifestSchema\` (\`packages/schemas/src/manifest.ts\`) records a single artifact-producing render call: git SHA, lockfile hash, codec, LLM call, request, artifact SHA, etc. It is **not** the right shape for a training run.

\`research/training/README.md\` describes a training-run receipt:

> Every training run emits, alongside the model checkpoint, a Wittgenstein manifest receipt:
> - Dataset hash (DVC-pinned)
> - Training step count + wall-clock + GPU type
> - Eval-metric snapshot (FID / CLIP / rFID per modality)
> - Git SHA of the harness at training time

That's a different shape entirely. It needs its own schema, validated, with cross-references to the published checkpoint asset.

## Why now (pre-training)

Currently nobody defines this schema. The first training run that produces a checkpoint will need to write a receipt the harness can later consume to bless that checkpoint into a decoder-family manifest. Defining the schema **before** the first training run avoids two competing receipt formats landing simultaneously.

## What to build

In \`packages/schemas/src/\` (likely a new \`training-manifest.ts\`):

\`\`\`ts
export const TrainingRunManifestSchema = z.object({
  schemaVersion: z.literal(\"witt.training.run-manifest/v0.1\"),
  runId: z.string().min(1),
  subprogram: z.enum([\"tokenizer\", \"adapter\", \"llm-head\"]),
  startedAt: z.string(),
  finishedAt: z.string().nullable(),

  harnessGitSha: z.string(),
  trainingCodeGitSha: z.string(),
  dockerImageSha: z.string().nullable(),
  dataset: z.object({
    dvcRev: z.string(),
    name: z.string(),
    sha256: z.string().regex(/^[a-f0-9]{64}$/),
  }),
  seed: z.number().int(),
  stepCount: z.number().int().nonnegative(),
  wallClockSec: z.number().nonnegative(),
  hardware: z.object({
    gpuModel: z.string(),
    gpuCount: z.number().int().positive(),
    nodeCount: z.number().int().positive(),
  }),
  evalSnapshots: z.array(/* per-modality metric receipts */),
  checkpoint: z.object({
    path: z.string(),
    sha256: z.string().regex(/^[a-f0-9]{64}$/),
    bytes: z.number().int().positive(),
    weightsLicense: z.enum([\"permissive\", \"research-only\"]),
  }),
});
\`\`\`

(Sketch — exact fields owned by the specialist.)

## Acceptance

- \`TrainingRunManifest\` type + zod schema exported from \`@wittgenstein/schemas\`.
- Round-trip test: parse a representative receipt fixture, assert no information loss.
- A cross-reference path: a \`DecoderFamilyManifestSchema\` (see #402 / #457) can point at one of these receipts as the asset's training provenance.
- Docs note on the relationship: TrainingRunManifest is the upstream evidence; DecoderFamilyManifest blesses a particular checkpoint as a decoder asset; RunManifest records an inference call that loads the decoder.

## Non-goals

- Do NOT couple the training schema to a specific framework (PyTorch / Lightning / etc.). Keep it framework-agnostic.
- Do NOT include training hyperparameters as a freeform blob inside the receipt — those live in the training code's own config.

## Refs

- Surfaced by the 2026-05-27 pre-training engineering sweep ([#481](https://github.com/p-to-q/wittgenstein/pull/481)).
- Related: [#400](https://github.com/p-to-q/wittgenstein/issues/400) DVC, [#399](https://github.com/p-to-q/wittgenstein/issues/399) Aim, [#402](https://github.com/p-to-q/wittgenstein/issues/402) decoder delivery, [#457](https://github.com/p-to-q/wittgenstein/pull/457) decoder audit delivery surface.
- This issue is **non-ML engineering scope** (schema design + validation), but lands adjacent to the ML specialist's training work.

**Comment by Jah-yee** [comment]
Sequencing note from the current M1/M1B pass:

This still looks like a worthwhile schema issue, but I would keep it **adjacent to** the training line rather than letting it blur into the current audit/bridge-delivery queue.

My current read after reviewing #457 / #491 / #492 / #493:
- `TrainingRunManifest` is useful and probably inevitable.
- It is **not** part of the minimum M1B audit-delivery closeout.
- The highest-value version of this issue is as a focused engineering follow-up once the training scaffold / owner-review line settles enough to tell us which fields are truly load-bearing versus aspirational.

So I would keep #482 open, but not promote it to a new umbrella. The existing M1B closeout umbrella is #474; this one should stay a narrow schema-design issue that absorbs lessons from #493 rather than trying to govern that whole line.

**Comment by koriyoshi2041** [comment]
Opened PR #541 to close this: https://github.com/p-to-q/wittgenstein/pull/541

What it lands:
- canonical `TrainingRunManifestSchema` + exported TypeScript types from `@wittgenstein/schemas`
- representative `witt.training.run-manifest/v0.1` fixture with round-trip/no-loss validation
- negative tests for freeform hyperparameter blobs and impossible run windows
- `TrainingRunManifestReferenceSchema` and `DecoderFamilyManifest` support for `assets.trainingProvenance`
- guard that decoder provenance checkpoint SHA must match the decoder asset `weightsSha256`
- docs clarifying the relationship between `TrainingRunManifest`, `DecoderFamilyManifest`, and inference `RunManifest`

Local validation passed across schemas, codec-image, full monorepo typecheck/test/build, dependency boundary lint, publish-surface lint, Prettier, and diff checks. The change intentionally does not touch the active tokenizer training files.

**Comment by koriyoshi2041** [comment]
Closed by merged PR #541: https://github.com/p-to-q/wittgenstein/pull/541

Merge commit: fa83a64d4f2499b718b097f3b867aa747421ad06

CI was green at merge time, including `Verify (Node + Python)`, static checks, reviewdog checks, Vercel, and CodeRabbit status. The merged contract adds a canonical `TrainingRunManifestSchema`, no-loss fixture test, strict rejection for freeform hyperparameter blobs, and a decoder-manifest provenance reference that must match the checkpoint SHA.
